### PR TITLE
presence ADR, presence on page navigation, full presence on connect

### PR DESCRIPTION
**Pull request by filipesilva**

### CODE REVIEW

**Review by neotyk — APPROVED** [review]

File: src/clj/athens/self_hosted/components/web.clj [hunk 1]
```diff
@@ -20,13 +20,10 @@
 
 (defn close-handler
   [channel status]
-  (let [username (clients/get-client-username channel)
-        ;; TODO: max-tx shouldn't be 42
-        presence-offline-event (athens.common-events/build-presence-offline-event 42 username)]
+  (let [username (clients/get-client-username channel)]
+    (presence/goodbye-handler channel)
     (clients/remove-client! channel)
```
Comment (neotyk) [inline src/clj/athens/self_hosted/components/web.clj:25]: We might want to `remove-client!` before calling `goodbye-handler` (because it's supposed to broadcast to all connected clients)

File: src/clj/athens/self_hosted/web/presence.clj [hunk 2]
```diff
@@ -37,6 +38,10 @@
       (clients/send! channel
                      (common-events/build-presence-all-online-event max-tx
                                                                     (clients/get-clients-usernames)))
+      (doseq [{:keys [username block-uid]} (vals @all-presence)]
+        (let [broadcast-presence-editing-event
+              (common-events/build-presence-broadcast-editing-event max-tx username block-uid)]
+          (clients/broadcast! broadcast-presence-editing-event)))
```
Comment (neotyk) [inline src/clj/athens/self_hosted/web/presence.clj:44]: This is too loud.
Did you maybe mean `clients/send!`?

Maybe you just wanted to send it to just connected client?
This happens already above.

File: src/clj/athens/self_hosted/web/presence.clj [hunk 3]
```diff
@@ -13,7 +14,7 @@
     (swap! max-id inc)))
 
 
-(defonce all-presence (ref []))
+(defonce all-presence (atom {}))
```
Comment (neotyk) [inline src/clj/athens/self_hosted/web/presence.clj:17]: You don't like `dosync` blocks?
Rick wink

File: src/clj/athens/self_hosted/web/presence.clj [hunk 4]
```diff
@@ -87,15 +94,23 @@
...
     (common-events/build-event-accepted id max-tx)))
 
 
 (defn goodbye-handler
-  [channel _event]
-  (let [_username (clients/get-client-username channel)]
-    (prn _event)))
+  [channel]
+  (let [username (clients/get-client-username channel)
+        ;; TODO: max-tx shouldn't be 42
```
Comment (neotyk) [inline src/clj/athens/self_hosted/web/presence.clj:109]: Presence events are forced to communicate `max-tx`,
this needs cleaning.
Maybe not part of this PR, but still need to clean it somewhere.

**Review by neotyk — CHANGES_REQUESTED** [review]

File: src/clj/athens/self_hosted/components/web.clj (hunk 1 above)
Comment (filipesilva) [reply src/clj/athens/self_hosted/components/web.clj:24]: Sgtm, done.

File: src/clj/athens/self_hosted/web/presence.clj (hunk 2 above)
Comment (filipesilva) [reply src/clj/athens/self_hosted/web/presence.clj]: Yes, I did mean that, changed now.

The connected client is getting the list of usernames above, but is not getting the current list of block presences.

File: src/clj/athens/self_hosted/web/presence.clj (hunk 3 above)
Comment (filipesilva) [reply src/clj/athens/self_hosted/web/presence.clj:17]: Just more used to atom semantics really.

File: src/clj/athens/self_hosted/web/presence.clj (hunk 4 above)
Comment (filipesilva) [reply src/clj/athens/self_hosted/web/presence.clj]: Fixed for now by actually determining the right one from datahike.

**Review by neotyk — APPROVED** [review]
